app/js/sendToChart.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * function to generate the data object for the chart displayed on the front end.
+ * @returns the data object for graph.js
+ */
+async function sendToChart() { 
+    let users = await createUsersObject();
+    users = searchAndFilter(users.data);
+    
+    let dataForChart = {};
+    let labels = await generateLabels(users);
+    let dubiousGrade = await generateDataset("Under 70%", labels, users, 0, 70, "#f27324");
+    let passingGrade = await generateDataset("70% to 97%", labels, users,  70, 97, "#94ba66");
+    let topGrade = await generateDataset("97% and above", labels, users, 97, 101, "#d2b4f9");
+
+    dataForChart.labels = labels;
+    dataForChart.datasets = [dubiousGrade, passingGrade, topGrade];
+    return dataForChart;
+};
+
+/**
+ * function to generate a dataset to be inserted into the chart's data object
+ * 
+ * @param {datasetLabel} string the label for the dataset
+ * @param {testLabels} Array the array of strings containing the names of the tests
+ * @param {users} Object the object containing information about all current users. Comes from createUsersObject
+ * @param {minPercentage} number the least percentage to be included in the range, inclusive
+ * @param {maxPercentage} number the greatest percentage to be included in the range, exclusive
+ * @param {color} string the colour code for entries in this dataset
+ * 
+ * @return Object a dataset object containing: A label, an array of data, and an array of background
+ *                colors for that data (all set to the same value, passed in as color) 
+ */
+async function generateDataset(datasetLabel, testLabels, users, minPercentage, maxPercentage, color) {
+
+    let dataset = {};
+    dataset.data = [];
+    dataset.label = datasetLabel;
+    dataset.backgroundColor = [];
```

Comment:
Can these properties be added to the object straight away when it's defined?

Reply:
Yes. Modified sendToChart and generateDataset to use this method of object assignment